### DISCUSSION TITLE
Analysing porosity with MTEX using SEM-BSE data embedded in EBSD data

### DISCUSSION THREAD

**Discussion by Remsdel** [mtex-toolbox/mtex#1213] [Ask Anything]
I am not sure this is possible and how so I need advice from the community:
In addition to my EBSD map, I have a SEM-BSE map with a 1-to-1 pixel correspondence to the EBSD map. Sample has a lot of porosity, inter and intragranular. The EBSD map already gives a good indication of the porosity using non-indexed pixels but the SEM-BSE map gives a much finer picture of the porosity. The latter image can be easily thresholded to obtain a binary mask. The goal is to apply the mask to the EBSD data and then use MTEX functions such as isInclusion to determined if porosity is inter of intra-granular and also get extra basic shape parameters (e.g. size) of the porosity 'Phase'. Applying ebsd(mask)=[ ] removes the porosity data from the dataset so does not allow to reconstruct 'grains' corresponding to the porosity in order to perform the analysis mentioned above. Is there another way such as defining ebsd(mask)=PorosityPhase that would allow to overwrite the underlying EBSD data that falls into the location of pores as given by the SEM-BSE binary mask?  Any advice is welcome, thanks in advance.

**Comment by nyyssont** [comment]
ebsd.CSList contains your phases and ebsd.phaseMap contains the phase ids. Both need to be updated to add a new phase.

Here is example code:
```
% Check from your import data file for a suitable crystalSymmetry to add to CSList:
ebsd.CSList{end+1} = ...
        crystalSymmetry('m-3m', [2.9 2.9 2.9],...
        'mineral', 'Pore', 'color', 'yellow')
  
% Update phaseMap:
ebsd.phaseMap(end+1) = ebsd.phaseMap(end) + 1;

% Change ebsd(ind) to last phase in phaseMap
ebsd(mask).phase = ebsd.phaseMap(end)
```

Now the last number in your phaseMap is a new phase called "Pore". It has a similar crystal structure to iron in this case but it could be anything.

I'm sure your data is complicated enough that manipulating CSList and phaseMap like this won't solve all your problems, but perhaps they are a step in the right direction?

**Reply by Remsdel** [reply]
Thanks, 
I tried your suggestion. It allows indeed to have pixels linked to porosity showed as a separate phase. However when reconstructing grains, using only indexed pixels, I get weird results, see picture attached. Yelllow phase corresponds to the porosity phase. Some 'porosity' grains are full of boundaries. I suppose there is some kind of interference with underlying non-indexed pixels which occurs often at the same location. Maybe ebsd(mask).phase = ebsd.phaseMap(end) is not enough to change all the properties of the concerned pixels? This is beyond my MTEX expertise at the moment. Any help welcome
![Reconstruction](https://user-images.githubusercontent.com/78489177/134139746-1a7f6855-a272-4cb2-9d3c-c22fec5068a3.PNG)
.

**Reply by nyyssont** [reply]
Are you interested in the orientation data of the indexed porous areas? If this information can be discarded, you can remove it from the pixels in MTEX as a preliminary processing step.

You could try:

```
% Remove rotation data from mask
ebsd(mask).rotations = quaternion.nan;

% Change phase
ebsd(mask).phase = ebsd.phaseMap(end);
```

edit: now that I look a bit more closely, I'm not sure this will help. I thought changing the phase of indexed pixels had messed up the grain reconstruction. But perhaps you can try anyway, shouldn't hurt.

**Reply by Remsdel** [reply]
Orientation is indeed not needed for porous area. Doing the step you just suggested leads to the opposite effect intended: all porous area now show a grid-like grain division whereas before some area did not show this effect. This makes me think that the contrary may be necessary: give an arbitrary orientation data to the indexed porous area? I don't know how this can be done.
Cordially,
Rémi

![Capture](https://user-images.githubusercontent.com/78489177/134155548-c5b794ef-95e7-425e-a9a3-fe6453f88bb5.PNG)
.

**Reply by nyyssont** [reply]
How about:

```
% Change phase
ebsd(mask).phase = ebsd.phaseMap(end);

% Give cube orientation to each measurement
ebsd(mask).rotations = orientation.cube(ebsd('Pore').CS);
```

If you wish, you can read here about creating various standard orientations:
https://mtex-toolbox.github.io/OrientationStandard.html

I'm a bit uncomfortable with assigning these dummy orientations to the orientation map... feels like a hack.

Perhaps someone else has a more elegant suggestion?

**Comment by Remsdel** [comment]
It does the trick in any case. I am willing to try another solution if somebody comes up with a less hacky way.
Thanks.
